Rudyanto

hactiv8rud opened the pull request.

hactiv8rud committed:
Revising API document to include register and login

Sursev07 commented on server/controllers/TodoController.js:
```diff
@@ -0,0 +1,154 @@
+                res.status(201).json(data);
+            })
+            .catch((err) => {
+                if (err.name === "SequelizeValidationError") {
```
nanti pindahin ke middleware/errorHandler yah 
abis lecture siang

Sursev07 commented on server/controllers/TodoController.js:
```diff
@@ -0,0 +1,154 @@
+        const obj = {
+            title: req.body.title,
+            description: req.body.description,
+            due_date: req.body.due_date
```
tambahin user id yah

Sursev07 commented on server/controllers/TodoController.js:
```diff
@@ -0,0 +1,154 @@
+        const id = Number(req.params.id);
+        Todo.findByPk(id)
+            .then((data) => {
+                if (data === null) {
```
```suggestion
                if (!data) {
```

Sursev07 commented on server/controllers/TodoController.js:
```diff
@@ -0,0 +1,154 @@
+                res.status(200).json(todo[1][0]);
+            }
+        } catch (err) {
+            if (err.name === "SequelizeValidationError") {
```
sama ya dengan komen diatas untuk errorhandler

Sursev07 commented on server/controllers/TodoController.js:
```diff
@@ -0,0 +1,154 @@
+                status: req.body.status,
+                due_date: req.body.due_date
+            }
+            const todo = await Todo.destroy({
```
penamaan variabel mungkin bisa diganti dengan result kali yah , 
karena hasilnya kan  menggambarkan delete success or not
```suggestion
            const result = await Todo.destroy({
```

Sursev07 commented on server/controllers/TodoController.js:
```diff
@@ -0,0 +1,154 @@
+            const obj = {
+                status: req.body.status,
+            }
+            const todo = await Todo.update(obj, {
```
penamaan variabel mungkin bisa diganti dengan result kali yah ,
karena hasilnya kan menggambarkan update success or not
```suggestion
            const result = await Todo.update(obj, {
```

Sursev07 commented on server/migrations/20201123121633-create-user.js:
```diff
@@ -0,0 +1,36 @@
+      last_name: {
+        type: Sequelize.STRING
+      },
+      email: {
```
unique:true

hactiv8rud committed:
Finishing documentation

Sursev07 commented on server/app.js:
```diff
@@ -0,0 +1,17 @@
+const PORT = 3000;
+const errorHandler = require("./middleware/errorHandler");
+const routes = require("./routes");
+
```
cors jangan lupa yah

Sursev07 commented on server/controllers/TodoController.js:
```diff
@@ -0,0 +1,101 @@
+
+    static async delete(req, res, next) {
+        try {
+            const id = Number(req.params.id);
```
misal delete nya gagal dia bakal tetep balikin 200?

Sursev07 commented on server/controllers/TodoController.js:
```diff
@@ -0,0 +1,101 @@
+            const obj = {
+                status: req.body.status,
+            }
+            const result = await Todo.update(obj, {
```
misal update nya gagal dia bakal tetep balikin 200?

hactiv8rud committed:
Added API doc

hactiv8rud committed:
Added onclick property to button on navbar

hactiv8rud committed:
Added Sweet Alert

hactiv8rud committed:
Added Search by category

hactiv8rud committed:
Merge pull request #1 from hactiv8rud/dev-add-filter

Added Search by category

hactiv8rud committed:
center the notification text and edit some layout

hactiv8rud committed:
fix typo in API doc

hactiv8rud committed:
deploy prep

hactiv8rud committed:
change url in ajax

hactiv8rud committed:
updated api doc

hactiv8rud committed:
update quote garden v3

hactiv8rud committed:
Merge pull request #2 from hactiv8rud/coba1

1. updated error handler
2. updated description input to textarea
3. updated layout for add form
4. updated quote garden v3

Commit:
fixed paths to css and images folders

Commit:
updated package-lock, config.json, updated to Google Identity Service Provider (not tested yet)

Commit:
fixed current weather by changing the api call, added /current-weather route, updated the variable assignment based on the structure of the api response

Commit:
changed the forecast to 5-day with 3-hour step from the free API call

Commit:
fixed getQuote API call, fixed src of todos icons, fixed the continuous call of getQuote api call, fixed get quote button layout so it appears inside the main div

Commit:
added helmet and noCache

Commit:
updated axios & jwt

Commit:
removed home button in navbar

Commit:
created migration model Tokens and added column for FK UserId to Tokens

Commit:
added attribute csrfToken to table Tokens

Commit:
added column to table Tokens in migration

Commit:
added association in model Token, edited an attribute in migration and in model for table Tokens from expiredAt to refreshTokenExpiredAt

Commit:
edited create-token in migration and user.js in model

Commit:
included env variables in api_doc.md, added cookie-parser, edited jwt.js to provide signing and verifying refresh token

Commit:
added getExpiredAt.js in helper, edited authentication.js in middlewa
re to improve security, refactored googleLogin in UserController.js from Promise to async-await, added csrf token in login, refreshToke
n, logout and googleLogin

Commit:
added error handling for expired session/jwt with status code 419

Commit:
added new routers in userRouters.js and edited term route to router

Commit:
updated auth with refresh token and csrf token, updated code to accomodate changes in ajax header variable naming rules, tested Sign in with Google

Commit:
edited Sign in with Google Button width

Commit:
fixed the authorization issue and logout

Commit:
removed google sign out outdated code in logout function in client

Commit:
added fullName in jwt signRefreshToken, sequelizerc, env variables for database config, updated api_doc.md to include edited env variables, sequelize-cli in dependencies

Commit:
uninstalled sequelize-cli

Commit:
edited index.js in models: changed config.json to config.js

Commit:
changed setting for deployment

Commit:
fixed undefined accesstoken, csrftoken in helper.js, included fullName in signRefreshToken in UserController.js

Commit:
removed onclick in home button and added function for home button in main.js

Commit:
added edited index.html

Commit:
replaced reauth function with showMainPage function in main.js

Commit:
improved web responsive design

Commit:
fixed typo in css and set 127.0.0.1:5500 to CORS origin

Commit:
added files app.js to commit

Commit:
fixed css .login-form min-width from rem to px

Commit:
fixed validation message from firstname to first name, edited layout for error messages to appear always fixed on top

Commit:
added title, close buttons in modal dialog

Commit:
added Loading GIF when loading weather forecast, added info modal

Commit:
updated helper.js to catch and display other type of error

Commit:
fixed code in helper.js to catch other type of error

Commit:
removed no-vertical-scroll class in CSS to enable pull to refresh in mobile view, added @media breakpoint for info modal to make sure that tech stack names are properly displayed

Commit:
added SVG to login and register page

Commit:
added more information in info section

Commit:
added more information in info section

Commit:
deleted unused files

Commit:
added more information in info section

Commit:
changed bootstrap 5 icon to bootstrap 4

Commit:
added loading animation

Commit:
edited typo in info

Commit:
edited typo in info

Commit:
edited margin bottom for validation error messages

Commit:
moved the fadeOut function for loadingMask to helper.js

Commit:
updated info section

Commit:
added github link in info section
